### PR TITLE
Add InstrumentedChannel

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java [hunk 1]
```diff
@@ -20,20 +20,26 @@
...
+            Collection<? extends Channel> channels,
+            UserAgent userAgent,
+            TaggedMetricRegistry metrics) {
         List<LimitedChannel> limitedChannels = channels.stream()
                 .map(ConcurrencyLimitedChannel::create)
                 .collect(toList());
 
         return new UserAgentChannel(
-                new RetryingChannel(new QueuedChannel(new RoundRobinChannel(limitedChannels))),
+                new InstrumentedChannel(
```
Comment (ellisjoe) [inline dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java]: think you want to stick this around the bottom most channel so we record just time observed in the remote server. then we can other metrics for things like time spent in the queue

File: dialogue-core/build.gradle [hunk 2]
```diff
@@ -9,6 +9,7 @@ dependencies {
     compile 'com.netflix.concurrency-limits:concurrency-limits-core'
     compile 'com.palantir.conjure.java.runtime:client-config'
     compile 'com.palantir.safe-logging:preconditions'
+    compile 'io.dropwizard.metrics:metrics-core'
```
Comment (ellisjoe) [inline dialogue-core/build.gradle:12]: pulling in tritium isn't sufficient?

File: dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java (hunk 1 above)
Comment (markelliot) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java]: we'd probably benefit from a method which flattens the channel initialization

Comment (uschi2000) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java]: can you elaborate, @markelliot ?

File: dialogue-core/build.gradle (hunk 2 above)
Comment (uschi2000) [reply dialogue-core/build.gradle:12]: fixed, thanks

File: dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java (hunk 1 above)
Comment (uschi2000) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java]: specifically, most of the channels have different arguments, so it's not clear to me how to canonicalize anything here. let me know.

File: dialogue-core/build.gradle (hunk 2 above)
Comment (schlosna) [reply dialogue-core/build.gradle:12]: ahh, I'm assuming this is what triggered https://github.com/palantir/tritium/issues/274

Comment (ellisjoe) [reply dialogue-core/build.gradle:12]: @schlosna yep!

**Pull request merged**